dictionaries/win32/src/generate.py

```diff
@@ -0,0 +1,168 @@
+import subprocess
+import sys
+import traceback
+import re
+import itertools
+import os
+
+from pycparser import c_ast, parse_file
+from pathlib import Path
+from platform import system
+from multiprocessing import Pool
+
+class DefNamesVisitor(c_ast.NodeVisitor):
+    decls = []
+
+    def add_name(self, name):
+        assert(name != None)
+        self.decls.append(name)
+
+    def visit_Decl(self, node):
+        if node.name != None:
+            self.add_name(node.name)
+
+    def visit_FuncDef(self, node):
+        self.add_name(node.decl.name)
+
+    def visit_Typedef(self, node):
+        self.add_name(node.name)
+
+    def visit_Struct(self, node):
+        self.add_name(node.name)
+
+        if node.decls != None:
+            for decl in node.decls:
+                if decl.name != None:
+                    self.add_name(decl.name)
+
+def flatten(array):
+    return list(itertools.chain.from_iterable(array))
+
+def expand(line):
+    # Do camelCase/PascalCase expansion
+    matches = re.findall(r'.+?(?:(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])|$)', line)
+    # Split snake_case strings
+    matches = flatten([term.split("_") for term in matches])
+    # Split around numbers
+    matches = flatten([re.split(r'[0-9]+', term) for term in matches])
+
+    return [term.lower() for term in matches if len(term) > 3 and not term.isnumeric()]
```

Comment:
It is better to keep the case. The spell checker is case sensitive.

Reply:
See https://github.com/streetsidesoftware/cspell-dicts/issues/589#issuecomment-945059578 for a discussion.